src/lib/Scenes/Map/MapRenderer.tsx

```diff
@@ -34,14 +35,35 @@ export const MapRenderer: React.SFC<{
         citySlug: props.citySlug,
         maxInt: MAX_GRAPHQL_INT,
       }}
-      render={({ props: mapProps }) => {
-        // TODO: Handle error, see LD-318.
-        if (mapProps || props.initialCoordinates) {
-          // viewer={null} is to handle the case where we want to render the map with initialCoordinates but the Relay
-          // response hasn't arrived yet. Relay requires us to pass an explicit null if the missing data is intentional.
-          return <GlobalMap viewer={null} {...mapProps} {...props} />
+      render={({ props: mapProps, error, retry }) => {
+        // viewer={null} is to handle the case where we want to render the map with initialCoordinates but the Relay
+        // response hasn't arrived yet. Relay requires us to pass an explicit null if the missing data is intentional.
+        const computedProps: any = { viewer: null, ...mapProps, ...props }
+
+        if (error) {
+          return (
+            <GlobalMap
+              {...computedProps}
+              viewer={null}
+              relayErrorState={{
+                error,
+                retry: () => {
+                  isRetrying = true
+                  retry()
+                },
+                isRetrying,
+              }}
+            />
+          )
+        } else if (isRetrying) {
+          isRetrying = false
+          return <GlobalMap {...computedProps} viewer={null} relayErrorState={{ isRetrying: true }} />
+        } else if (mapProps || props.initialCoordinates) {
+          return <GlobalMap {...computedProps} />
+        } else {
```

Comment:
I need to come back and comment some of this – I had to learn a lot about how query renderers work and would like to capture that knowledge.